cmd/ndm_daemonset/filter/osdiskexcludefilter.go

```diff
@@ -109,7 +113,18 @@ func (odf *oSDiskExcludeFilter) Include(d *controller.DiskInfo) bool {
 	return true
 }
 
-// Exclude returns true if disk devpath matches with excludeDevPath
+// Exclude returns true if disk devpath does not match with excludeDevPath
 func (odf *oSDiskExcludeFilter) Exclude(d *controller.DiskInfo) bool {
-	return odf.excludeDevPath != d.Path
+	// The partitionRegex is chosen depending on whether the device uses
+	// the p[0-9] partition naming structure or not.
+	var partitionRegex string
+	if util.MatchRegex(".+[0-9]+$", odf.excludeDevPath) {
+		// matches loop0, loop0p1, nvme3n0p1
+		partitionRegex = "(p[0-9]+)?$"
+	} else {
+		// matches sda, sda1
+		partitionRegex = "[0-9]*$"
+	}
+	glog.Info("applying regex ", odf.excludeDevPath+partitionRegex, " for os-disk-exclude-filter")
```

Comment:
log can be improved by including the d.Path itself.

Reply:
Done